torch/nn/modules/module.py

```diff
@@ -565,6 +673,12 @@ def _slow_forward(self, *input, **kwargs):
         return result
 
     def __call__(self, *input, **kwargs):
+        for hook in self._global_forward_pre_hooks.values():
```

Comment:
This could be merged in the for loop below to re-use the code by chaining the two iterators.

Same for the two others below.